File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        ArrayList<int[]> res = new ArrayList<>();
        int n= intervals.length;
        if(intervals.length<=1){
            return intervals;
        }
        Arrays.sort(intervals,(a,b)-> Integer.compare(a[0],b[0]));

        int[] current =intervals[0];
        for(int i=1;i<n;i++){
        
            if(current[1]>= intervals[i][0]){
                current[1] = Math.max(current[1],intervals[i][1]);
               
            }
            else{
              res.add(current);
                current=intervals[i];
            }
           

        }  res.add(current);
         return res.toArray(new int[res.size()][]);

    }
}

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution(object):
    def merge(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: List[List[int]]
        """

        merged = []
        n = len(intervals)
        i = 0
        intervals.sort(key=lambda x: x[0])


        oldinterval = intervals[i] 

        while i < n :
            if oldinterval[1] >= intervals[i][0] :
                oldinterval[1] = max(intervals[i][1] , oldinterval[1] )
                i += 1 
            else :
                merged.append(oldinterval)
                oldinterval = intervals[i]
                
                i += 1   
        
        if i == n :
           merged.append(oldinterval) 
           

        
        return merged